core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -42,10 +43,12 @@ import scala.concurrent.Future
  *                                     |                      |
  *                                     +----------------------+
  */
-object Scheduler {
+object Scheduler extends StrictLogging {
+
   case class MesosConnection(mesosHostName: String)
 
   def connect(mesosHostName: String): Future[(MesosConnection, Flow[SpecEvent, StateEvent, NotUsed])] = {
+    logger.info(s"Connecting to {}", kv("mesosHostName", mesosHostName))
```

Comment:
@alenkacz and I had a brief discussion on the format. Here is the [slack thread](https://mesosphere.slack.com/archives/G7Y5Q786S/p1548922017700400). This is what I really liked
> @kjeschkies yeah, I am open to pretty much anything. What I like about the fluent approach is that you can basically inject some metadata into logger with `logger.with` and then pass that logger into some method (so you don't have to have the whole metadata at hand when you're printing your message)

So this should be something like
```scala
logger.with("meosHostName" -> mesosHostName).info("Connecting")
```

which would output something like
```json
{"class": "Scheduler", "context": {"mesosHostName": "localhost"}, "message": "conntecting"}
```

What would the output look like in your example?

Reply:
yeah sorry, I did not finish writing the doc for that, I thought I'll do it when I come back from my vacation. But most of the ideas are in that thread (plus a little bit in my head). I can write it down since there are several options how to do that... (but I also like the one @jeschkies is describing, that's the one I saw in the hashicorps' golang library for logging).